Adding console scripts

andersoncarlosfs opened the pull request:
- adding the scripts under the python package
- renaming the scripts
- creating main functions

![199483762-870fb770-144d-42a3-af60-1342269fcea7](https://user-images.githubusercontent.com/23130084/199488857-d76a1424-de9f-4a56-a66a-e2f1fbbc31f8.png)

tomsercu commented on setup.py:
```diff
@@ -34,8 +42,15 @@
     zip_safe=True,
+    entry_points={
+        "console_scripts": [
+            "esmfold_extract=esm.scripts.esmfold_extract:main",
+            "esmfold_inference=esm.scripts.esmfold_inference:main",
```
```suggestion
            "esm-extract=esm.scripts.extract:main",
            "esm-fold=esm.scripts.fold:main",
```

tomsercu commented on setup.py:
```diff
@@ -25,6 +25,14 @@
+    "esm.model": "esm/model",
+    "esm.inverse_folding": "esm/inverse_folding",
+    "esm.esmfold.v1": "esm/esmfold/v1",
+    "esm.scripts": "scripts"
```
@nikitos9000 shall we move `scripts` under `esm` to make things more consistent? Only we don't want to move `scripts/atlas` directory; we could move `atlas` to be at root level..
Or can we just not have `esm.scripts` in the `sources`, it seems unnecessary for the fairseq cli https://github.com/facebookresearch/fairseq/blob/main/setup.py

andersoncarlosfs replied:
I don't have any opinion... As I  have free time now, just let me know your preference...

andersoncarlosfs replied:
PS: without `esm.scripts` in `sources`, pip/python will be unable to find the cli scripts

tomsercu reviewed (changes requested):
Hi @andersoncarlosfs thanks for submitting this PR!
The goal of this PR is adding a CLI to the esm repo.
This makes sense, I like it.
A couple of things to point out:
* `scripts/extract.py → scripts/esmfold_extract.py` this is incorrect as the script allows extracting embeddings from ESM, not ESMFold.
* Instead let's rename `scripts/esmfold_inference.py` to `fold.py`, and make the CLI names `esm-extract` and `esm-fold` consistent with script names.
* Document CLI in README

Let us know if you can address those! And tagging @nikitos9000 to help support where needed and for a second look

andersoncarlosfs commented:
> Hi @andersoncarlosfs thanks for submitting this PR! The goal of this PR is adding a CLI to the esm repo. This makes sense, I like it. A couple of things to point out:
> 
>     * `scripts/extract.py → scripts/esmfold_extract.py` this is incorrect as the script allows extracting embeddings from ESM, not ESMFold.
> 
>     * Instead let's rename `scripts/esmfold_inference.py` to `fold.py`, and make the CLI names `esm-extract` and `esm-fold` consistent with script names.
> 
>     * Document CLI in README
> 
> 
> Let us know if you can address those! And tagging @nikitos9000 to help support where needed and for a second look

Hello @tomsercu and @nikitos9000,

I updated the code and some notes in README following your recommendation. I would also inform you that I changed the environment.yml to use a requirements.txt, which can be part of the setup.py as requirement (listing the dependencies). 
<img width="976" alt="esm-extract" src="https://user-images.githubusercontent.com/23130084/201943554-7874b715-d55f-47dd-b5e6-44fea5685689.png">
<img width="976" alt="esm-fold" src="https://user-images.githubusercontent.com/23130084/201943562-5a44e0f8-6a2f-48b6-be67-a0e2faaf1f7b.png">

```
$ conda env create --name test --file=environment.yml
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Installing pip dependencies: ...working... Ran pip subprocess with arguments:
['/miniconda/envs/test/bin/python', '-m', 'pip', 'install', '-U', '-r', '/esm/condaenv.8db1ukvk.requirements.txt']
Pip subprocess output:
Collecting DLLogger@ git+https://github.com/NVIDIA/dllogger#egg=dllogger
  Cloning https://github.com/NVIDIA/dllogger to /private/var/folders/07/jg8wgn3j17b7nwbp0z668t_c0000gn/T/pip-install-o_eo3b9l/dllogger_a50df175eb114c91a68b82761b0bfafd
  Resolved https://github.com/NVIDIA/dllogger to commit 0540a43971f4a8a16693a9de9de73c1072020769
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting pytest
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pandas
  Downloading pandas-1.3.5-cp37-cp37m-macosx_10_9_x86_64.whl (11.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.0/11.0 MB 23.8 MB/s eta 0:00:00
Collecting biopython==1.79
  Using cached biopython-1.79-cp37-cp37m-macosx_10_9_x86_64.whl (2.3 MB)
Collecting deepspeed==0.5.9
  Using cached deepspeed-0.5.9-py3-none-any.whl
Collecting dm-tree==0.1.6
  Using cached dm_tree-0.1.6-cp37-cp37m-macosx_10_14_x86_64.whl (94 kB)
Collecting ml-collections==0.1.0
  Using cached ml_collections-0.1.0-py3-none-any.whl (88 kB)
Collecting numpy==1.21.2
  Using cached numpy-1.21.2-cp37-cp37m-macosx_10_9_x86_64.whl (16.9 MB)
Collecting PyYAML==5.4.1
  Using cached PyYAML-5.4.1-cp37-cp37m-macosx_10_9_x86_64.whl (249 kB)
Collecting requests==2.26.0
  Using cached requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting scipy==1.7.1
  Using cached scipy-1.7.1-cp37-cp37m-macosx_10_9_x86_64.whl (32.6 MB)
Collecting tqdm==4.62.2
  Using cached tqdm-4.62.2-py2.py3-none-any.whl (76 kB)
Collecting typing-extensions==3.10.0.2
  Using cached typing_extensions-3.10.0.2-py3-none-any.whl (26 kB)
Collecting pytorch_lightning==1.5.10
  Using cached pytorch_lightning-1.5.10-py3-none-any.whl (527 kB)
Collecting wandb==0.12.21
  Using cached wandb-0.12.21-py2.py3-none-any.whl (1.8 MB)
Requirement already satisfied: torch in /miniconda/envs/test/lib/python3.7/site-packages (from deepspeed==0.5.9->-r /esm/requirements.txt (line 4)) (1.12.1)
Requirement already satisfied: packaging in /miniconda/envs/test/lib/python3.7/site-packages (from deepspeed==0.5.9->-r /esm/requirements.txt (line 4)) (21.3)
Collecting hjson
  Using cached hjson-3.1.0-py3-none-any.whl (54 kB)
Collecting psutil
  Using cached psutil-5.9.4-cp36-abi3-macosx_10_9_x86_64.whl (243 kB)
Collecting ninja
  Using cached ninja-1.11.1-py2.py3-none-macosx_10_9_universal2.macosx_10_9_x86_64.macosx_11_0_arm64.macosx_11_0_universal2.whl (270 kB)
Collecting py-cpuinfo
  Using cached py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)
Requirement already satisfied: six>=1.12.0 in /miniconda/envs/test/lib/python3.7/site-packages (from dm-tree==0.1.6->-r /esm/requirements.txt (line 5)) (1.16.0)
Collecting contextlib2
  Using cached contextlib2-21.6.0-py2.py3-none-any.whl (13 kB)
Collecting absl-py
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting tensorboard>=2.2.0
  Using cached tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
Collecting fsspec[http]!=2021.06.0,>=2021.05.0
  Using cached fsspec-2022.11.0-py3-none-any.whl (139 kB)
Collecting future>=0.17.1
  Using cached future-0.18.2-py3-none-any.whl
Collecting torchmetrics>=0.4.1
  Using cached torchmetrics-0.10.2-py3-none-any.whl (529 kB)
Requirement already satisfied: setuptools==59.5.0 in /miniconda/envs/test/lib/python3.7/site-packages (from pytorch_lightning==1.5.10->-r /esm/requirements.txt (line 13)) (59.5.0)
Collecting pyDeprecate==0.3.1
  Using cached pyDeprecate-0.3.1-py3-none-any.whl (10 kB)
Collecting setproctitle
  Using cached setproctitle-1.3.2-cp37-cp37m-macosx_10_9_x86_64.whl (11 kB)
Collecting promise<3,>=2.0
  Using cached promise-2.3-py3-none-any.whl
Collecting docker-pycreds>=0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting GitPython>=1.0.0
  Using cached GitPython-3.1.29-py3-none-any.whl (182 kB)
Collecting pathtools
  Using cached pathtools-0.1.2-py3-none-any.whl
Requirement already satisfied: Click!=8.0.0,>=7.0 in /.local/lib/python3.7/site-packages (from wandb==0.12.21->-r /esm/requirements.txt (line 14)) (7.0)
Collecting protobuf<4.0dev,>=3.12.0
  Using cached protobuf-3.20.3-cp37-cp37m-macosx_10_9_x86_64.whl (981 kB)
Collecting sentry-sdk>=1.0.0
  Using cached sentry_sdk-1.11.0-py2.py3-none-any.whl (168 kB)
Collecting shortuuid>=0.5.0
  Using cached shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting attrs>=19.2.0
  Downloading attrs-22.1.0-py2.py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 1.7 MB/s eta 0:00:00
Collecting importlib-metadata>=0.12
  Downloading importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting pytz>=2017.3
  Downloading pytz-2022.6-py2.py3-none-any.whl (498 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 498.1/498.1 kB 10.5 MB/s eta 0:00:00
Collecting python-dateutil>=2.7.3
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting aiohttp!=4.0.0a0,!=4.0.0a1
  Using cached aiohttp-3.8.3-cp37-cp37m-macosx_10_9_x86_64.whl (355 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.9-py3-none-any.whl (63 kB)
Requirement already satisfied: zipp>=0.5 in /miniconda/envs/test/lib/python3.7/site-packages (from importlib-metadata>=0.12->pytest->-r /esm/requirements.txt (line 1)) (3.10.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /miniconda/envs/test/lib/python3.7/site-packages (from packaging->deepspeed==0.5.9->-r /esm/requirements.txt (line 4)) (3.0.9)
Collecting werkzeug>=1.0.1
  Using cached Werkzeug-2.2.2-py3-none-any.whl (232 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting google-auth<3,>=1.6.3
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-macosx_10_9_x86_64.whl (3.5 MB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
Collecting grpcio>=1.24.3
  Using cached grpcio-1.50.0-cp37-cp37m-macosx_10_10_x86_64.whl (4.5 MB)
Collecting markdown>=2.6.8
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Requirement already satisfied: wheel>=0.26 in /miniconda/envs/test/lib/python3.7/site-packages (from tensorboard>=2.2.0->pytorch_lightning==1.5.10->-r /esm/requirements.txt (line 13)) (0.38.4)
Collecting aiosignal>=1.1.2
  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting frozenlist>=1.1.1
  Using cached frozenlist-1.3.3-cp37-cp37m-macosx_10_9_x86_64.whl (36 kB)
Collecting multidict<7.0,>=4.5
  Using cached multidict-6.0.2-cp37-cp37m-macosx_10_9_x86_64.whl (28 kB)
Collecting yarl<2.0,>=1.0
  Using cached yarl-1.8.1-cp37-cp37m-macosx_10_9_x86_64.whl (60 kB)
Collecting asynctest==0.13.0
  Using cached asynctest-0.13.0-py3-none-any.whl (26 kB)
Collecting async-timeout<5.0,>=4.0.0a3
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting MarkupSafe>=2.1.1
  Using cached MarkupSafe-2.1.1-cp37-cp37m-macosx_10_9_x86_64.whl (13 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Building wheels for collected packages: DLLogger
  Building wheel for DLLogger (setup.py): started
  Building wheel for DLLogger (setup.py): finished with status 'done'
  Created wheel for DLLogger: filename=DLLogger-1.0.0-py3-none-any.whl size=5670 sha256=bd6190888b15c08ebb8257b3f0376b575ac4b79274390b9b01230b6c58e096fa
  Stored in directory: /private/var/folders/07/jg8wgn3j17b7nwbp0z668t_c0000gn/T/pip-ephem-wheel-cache-z2h_6dx7/wheels/f6/0b/e6/d9566189ca2a94642aa7060bbf15c51ec8c464053945b1358b
Successfully built DLLogger
Installing collected packages: typing-extensions, tensorboard-plugin-wit, pytz, pyasn1, py-cpuinfo, pathtools, ninja, iniconfig, hjson, urllib3, tqdm, tomli, tensorboard-data-server, smmap, shortuuid, setproctitle, rsa, PyYAML, python-dateutil, pyDeprecate, pyasn1-modules, psutil, protobuf, promise, oauthlib, numpy, multidict, MarkupSafe, importlib-metadata, idna, grpcio, future, fsspec, frozenlist, exceptiongroup, docker-pycreds, dm-tree, DLLogger, contextlib2, charset-normalizer, certifi, cachetools, attrs, asynctest, async-timeout, absl-py, yarl, werkzeug, torchmetrics, sentry-sdk, scipy, requests, pluggy, pandas, ml-collections, markdown, google-auth, gitdb, deepspeed, biopython, aiosignal, requests-oauthlib, pytest, GitPython, aiohttp, wandb, google-auth-oauthlib, tensorboard, pytorch_lightning
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.4.0
    Uninstalling typing_extensions-4.4.0:
      Successfully uninstalled typing_extensions-4.4.0
  Attempting uninstall: PyYAML
    Found existing installation: PyYAML 6.0
    Uninstalling PyYAML-6.0:
      Successfully uninstalled PyYAML-6.0
  Attempting uninstall: numpy
    Found existing installation: numpy 1.21.5
    Uninstalling numpy-1.21.5:
      Successfully uninstalled numpy-1.21.5
Successfully installed DLLogger-1.0.0 GitPython-3.1.29 MarkupSafe-2.1.1 PyYAML-5.4.1 absl-py-1.3.0 aiohttp-3.8.3 aiosignal-1.3.1 async-timeout-4.0.2 asynctest-0.13.0 attrs-22.1.0 biopython-1.79 cachetools-5.2.0 certifi-2022.9.24 charset-normalizer-2.0.12 contextlib2-21.6.0 deepspeed-0.5.9 dm-tree-0.1.6 docker-pycreds-0.4.0 exceptiongroup-1.0.4 frozenlist-1.3.3 fsspec-2022.11.0 future-0.18.2 gitdb-4.0.9 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 hjson-3.1.0 idna-3.4 importlib-metadata-5.0.0 iniconfig-1.1.1 markdown-3.4.1 ml-collections-0.1.0 multidict-6.0.2 ninja-1.11.1 numpy-1.21.2 oauthlib-3.2.2 pandas-1.3.5 pathtools-0.1.2 pluggy-1.0.0 promise-2.3 protobuf-3.20.3 psutil-5.9.4 py-cpuinfo-9.0.0 pyDeprecate-0.3.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 pytest-7.2.0 python-dateutil-2.8.2 pytorch_lightning-1.5.10 pytz-2022.6 requests-2.26.0 requests-oauthlib-1.3.1 rsa-4.9 scipy-1.7.1 sentry-sdk-1.11.0 setproctitle-1.3.2 shortuuid-1.0.11 smmap-5.0.0 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 tomli-2.0.1 torchmetrics-0.10.2 tqdm-4.62.2 typing-extensions-3.10.0.2 urllib3-1.26.12 wandb-0.12.21 werkzeug-2.2.2 yarl-1.8.1

done
#
# To activate this environment, use
#
#     $ conda activate test
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```
```
$ pip install ./
Processing /esm
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting DLLogger@ git+https://github.com/NVIDIA/dllogger#egg=dllogger
  Cloning https://github.com/NVIDIA/dllogger to /private/var/folders/07/jg8wgn3j17b7nwbp0z668t_c0000gn/T/pip-install-d57kfs4j/dllogger_2df039ccbdae49f3b01a7368a5aba1e6
  Resolved https://github.com/NVIDIA/dllogger to commit 0540a43971f4a8a16693a9de9de73c1072020769
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting numpy==1.21.2
  Using cached numpy-1.21.2-cp37-cp37m-macosx_10_9_x86_64.whl (16.9 MB)
Collecting dm-tree==0.1.6
  Using cached dm_tree-0.1.6-cp37-cp37m-macosx_10_14_x86_64.whl (94 kB)
Collecting deepspeed==0.5.9
  Using cached deepspeed-0.5.9.tar.gz (510 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting pytest
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pytorch-lightning==1.5.10
  Using cached pytorch_lightning-1.5.10-py3-none-any.whl (527 kB)
Collecting wandb==0.12.21
  Using cached wandb-0.12.21-py2.py3-none-any.whl (1.8 MB)
Collecting biopython==1.79
  Using cached biopython-1.79-cp37-cp37m-macosx_10_9_x86_64.whl (2.3 MB)
Collecting tqdm==4.62.2
  Using cached tqdm-4.62.2-py2.py3-none-any.whl (76 kB)
Collecting typing-extensions==3.10.0.2
  Using cached typing_extensions-3.10.0.2-py3-none-any.whl (26 kB)
Collecting pandas
  Using cached pandas-1.3.5-cp37-cp37m-macosx_10_9_x86_64.whl (11.0 MB)
Collecting PyYAML==5.4.1
  Using cached PyYAML-5.4.1-cp37-cp37m-macosx_10_9_x86_64.whl (249 kB)
Collecting ml-collections==0.1.0
  Using cached ml_collections-0.1.0-py3-none-any.whl (88 kB)
Collecting requests==2.26.0
  Using cached requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting scipy==1.7.1
  Using cached scipy-1.7.1-cp37-cp37m-macosx_10_9_x86_64.whl (32.6 MB)
Collecting hjson
  Using cached hjson-3.1.0-py3-none-any.whl (54 kB)
Collecting ninja
  Using cached ninja-1.11.1-py2.py3-none-macosx_10_9_universal2.macosx_10_9_x86_64.macosx_11_0_arm64.macosx_11_0_universal2.whl (270 kB)
Collecting packaging
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 kB 1.1 MB/s eta 0:00:00
Collecting psutil
  Using cached psutil-5.9.4-cp36-abi3-macosx_10_9_x86_64.whl (243 kB)
Collecting py-cpuinfo
  Using cached py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)
Collecting torch
  Downloading torch-1.13.0-cp37-none-macosx_10_9_x86_64.whl (137.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 137.5/137.5 MB 15.7 MB/s eta 0:00:00
Collecting six>=1.12.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting absl-py
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting contextlib2
  Using cached contextlib2-21.6.0-py2.py3-none-any.whl (13 kB)
Collecting pyDeprecate==0.3.1
  Using cached pyDeprecate-0.3.1-py3-none-any.whl (10 kB)
Collecting future>=0.17.1
  Using cached future-0.18.2.tar.gz (829 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting torchmetrics>=0.4.1
  Using cached torchmetrics-0.10.2-py3-none-any.whl (529 kB)
Collecting setuptools==59.5.0
  Using cached setuptools-59.5.0-py3-none-any.whl (952 kB)
Collecting fsspec[http]!=2021.06.0,>=2021.05.0
  Using cached fsspec-2022.11.0-py3-none-any.whl (139 kB)
Collecting tensorboard>=2.2.0
  Using cached tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Requirement already satisfied: certifi>=2017.4.17 in /miniconda/envs/test/lib/python3.7/site-packages (from requests==2.26.0->fair-esm==2.0.0) (2022.9.24)
Collecting setproctitle
  Using cached setproctitle-1.3.2-cp37-cp37m-macosx_10_9_x86_64.whl (11 kB)
Collecting pathtools
  Using cached pathtools-0.1.2.tar.gz (11 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting sentry-sdk>=1.0.0
  Using cached sentry_sdk-1.11.0-py2.py3-none-any.whl (168 kB)
Collecting protobuf<4.0dev,>=3.12.0
  Using cached protobuf-3.20.3-cp37-cp37m-macosx_10_9_x86_64.whl (981 kB)
Collecting docker-pycreds>=0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting promise<3,>=2.0
  Using cached promise-2.3.tar.gz (19 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Requirement already satisfied: Click!=8.0.0,>=7.0 in /.local/lib/python3.7/site-packages (from wandb==0.12.21->fair-esm==2.0.0) (7.0)
Collecting shortuuid>=0.5.0
  Using cached shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting GitPython>=1.0.0
  Using cached GitPython-3.1.29-py3-none-any.whl (182 kB)
Collecting python-dateutil>=2.7.3
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pytz>=2017.3
  Using cached pytz-2022.6-py2.py3-none-any.whl (498 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting attrs>=19.2.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting importlib-metadata>=0.12
  Using cached importlib_metadata-5.0.0-py3-none-any.whl (21 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting aiohttp!=4.0.0a0,!=4.0.0a1
  Using cached aiohttp-3.8.3-cp37-cp37m-macosx_10_9_x86_64.whl (355 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.9-py3-none-any.whl (63 kB)
Collecting zipp>=0.5
  Downloading zipp-3.10.0-py3-none-any.whl (6.2 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.3/98.3 kB 2.7 MB/s eta 0:00:00
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-macosx_10_9_x86_64.whl (3.5 MB)
Collecting werkzeug>=1.0.1
  Using cached Werkzeug-2.2.2-py3-none-any.whl (232 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
Requirement already satisfied: wheel>=0.26 in /miniconda/envs/test/lib/python3.7/site-packages (from tensorboard>=2.2.0->pytorch-lightning==1.5.10->fair-esm==2.0.0) (0.37.1)
Collecting google-auth<3,>=1.6.3
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting markdown>=2.6.8
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting grpcio>=1.24.3
  Using cached grpcio-1.50.0-cp37-cp37m-macosx_10_10_x86_64.whl (4.5 MB)
Collecting multidict<7.0,>=4.5
  Using cached multidict-6.0.2-cp37-cp37m-macosx_10_9_x86_64.whl (28 kB)
Collecting frozenlist>=1.1.1
  Using cached frozenlist-1.3.3-cp37-cp37m-macosx_10_9_x86_64.whl (36 kB)
Collecting async-timeout<5.0,>=4.0.0a3
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting aiosignal>=1.1.2
  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Collecting yarl<2.0,>=1.0
  Using cached yarl-1.8.1-cp37-cp37m-macosx_10_9_x86_64.whl (60 kB)
Collecting asynctest==0.13.0
  Using cached asynctest-0.13.0-py3-none-any.whl (26 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting MarkupSafe>=2.1.1
  Using cached MarkupSafe-2.1.1-cp37-cp37m-macosx_10_9_x86_64.whl (13 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Building wheels for collected packages: fair-esm, deepspeed, DLLogger, future, promise, pathtools
  Building wheel for fair-esm (pyproject.toml): started
  Building wheel for fair-esm (pyproject.toml): finished with status 'done'
  Created wheel for fair-esm: filename=fair_esm-2.0.0-py3-none-any.whl size=95768 sha256=7ab6257449593ce81f146e7f5c2726662e7adb7d6e32d86d978a9354e923d640
  Stored in directory: /private/var/folders/07/jg8wgn3j17b7nwbp0z668t_c0000gn/T/pip-ephem-wheel-cache-a1z1hvm_/wheels/91/ee/2e/5b6df28730d23901440d956c96288b44c5cc39d59c4662c60f
  Building wheel for deepspeed (setup.py): started
  Building wheel for deepspeed (setup.py): finished with status 'done'
  Created wheel for deepspeed: filename=deepspeed-0.5.9-py3-none-any.whl size=524278 sha256=4fde4e24231417fbd07adfd4d769630c5fb428138172284387296a0f90ed739e
  Stored in directory: /Library/Caches/pip/wheels/06/00/64/2aeb9e77c46196d95544ae3c384735c2a819c6f101992c76ab
  Building wheel for DLLogger (setup.py): started
  Building wheel for DLLogger (setup.py): finished with status 'done'
  Created wheel for DLLogger: filename=DLLogger-1.0.0-py3-none-any.whl size=5653 sha256=cee738a4cf4e33b73440c7f46f9b5023c22346fcba08d9f36de59c543c2ebaf5
  Stored in directory: /private/var/folders/07/jg8wgn3j17b7nwbp0z668t_c0000gn/T/pip-ephem-wheel-cache-a1z1hvm_/wheels/f6/0b/e6/d9566189ca2a94642aa7060bbf15c51ec8c464053945b1358b
  Building wheel for future (setup.py): started
  Building wheel for future (setup.py): finished with status 'done'
  Created wheel for future: filename=future-0.18.2-py3-none-any.whl size=491058 sha256=7d346056a353a4f024a6796ed8d16bc95d139ab0f7a8ce89a57550a148cd7a9d
  Stored in directory: /Library/Caches/pip/wheels/56/b0/fe/4410d17b32f1f0c3cf54cdfb2bc04d7b4b8f4ae377e2229ba0
  Building wheel for promise (setup.py): started
  Building wheel for promise (setup.py): finished with status 'done'
  Created wheel for promise: filename=promise-2.3-py3-none-any.whl size=21486 sha256=299e6a006cfaad16fc882061af7f4be9be03c32b589ce989837bfa652cd1214d
  Stored in directory: /Library/Caches/pip/wheels/29/93/c6/762e359f8cb6a5b69c72235d798804cae523bbe41c2aa8333d
  Building wheel for pathtools (setup.py): started
  Building wheel for pathtools (setup.py): finished with status 'done'
  Created wheel for pathtools: filename=pathtools-0.1.2-py3-none-any.whl size=8792 sha256=9bc76eebd1b1528f0bf2fd98900e91e093324ede92a9e0071bba8ded07d1f889
  Stored in directory: /Library/Caches/pip/wheels/3e/31/09/fa59cef12cdcfecc627b3d24273699f390e71828921b2cbba2
Successfully built fair-esm deepspeed DLLogger future promise pathtools
Installing collected packages: typing-extensions, tensorboard-plugin-wit, pytz, pyasn1, py-cpuinfo, pathtools, ninja, iniconfig, hjson, zipp, urllib3, tqdm, torch, tomli, tensorboard-data-server, smmap, six, shortuuid, setuptools, setproctitle, rsa, PyYAML, pyparsing, pyDeprecate, pyasn1-modules, psutil, protobuf, oauthlib, numpy, multidict, MarkupSafe, idna, future, fsspec, frozenlist, exceptiongroup, DLLogger, contextlib2, charset-normalizer, cachetools, attrs, asynctest, async-timeout, absl-py, yarl, werkzeug, sentry-sdk, scipy, requests, python-dateutil, promise, packaging, ml-collections, importlib-metadata, grpcio, google-auth, gitdb, docker-pycreds, dm-tree, biopython, aiosignal, torchmetrics, requests-oauthlib, pluggy, pandas, markdown, GitPython, deepspeed, aiohttp, wandb, pytest, google-auth-oauthlib, tensorboard, pytorch-lightning, fair-esm
  Attempting uninstall: setuptools
    Found existing installation: setuptools 65.5.0
    Uninstalling setuptools-65.5.0:
      Successfully uninstalled setuptools-65.5.0
Successfully installed DLLogger-1.0.0 GitPython-3.1.29 MarkupSafe-2.1.1 PyYAML-5.4.1 absl-py-1.3.0 aiohttp-3.8.3 aiosignal-1.3.1 async-timeout-4.0.2 asynctest-0.13.0 attrs-22.1.0 biopython-1.79 cachetools-5.2.0 charset-normalizer-2.0.12 contextlib2-21.6.0 deepspeed-0.5.9 dm-tree-0.1.6 docker-pycreds-0.4.0 exceptiongroup-1.0.4 fair-esm-2.0.0 frozenlist-1.3.3 fsspec-2022.11.0 future-0.18.2 gitdb-4.0.9 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 hjson-3.1.0 idna-3.4 importlib-metadata-5.0.0 iniconfig-1.1.1 markdown-3.4.1 ml-collections-0.1.0 multidict-6.0.2 ninja-1.11.1 numpy-1.21.2 oauthlib-3.2.2 packaging-21.3 pandas-1.3.5 pathtools-0.1.2 pluggy-1.0.0 promise-2.3 protobuf-3.20.3 psutil-5.9.4 py-cpuinfo-9.0.0 pyDeprecate-0.3.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 pyparsing-3.0.9 pytest-7.2.0 python-dateutil-2.8.2 pytorch-lightning-1.5.10 pytz-2022.6 requests-2.26.0 requests-oauthlib-1.3.1 rsa-4.9 scipy-1.7.1 sentry-sdk-1.11.0 setproctitle-1.3.2 setuptools-59.5.0 shortuuid-1.0.11 six-1.16.0 smmap-5.0.0 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 tomli-2.0.1 torch-1.13.0 torchmetrics-0.10.2 tqdm-4.62.2 typing-extensions-3.10.0.2 urllib3-1.26.12 wandb-0.12.21 werkzeug-2.2.2 yarl-1.8.1 zipp-3.10.0
```

tomsercu commented on requirements.txt:
```diff
@@ -0,0 +1,15 @@
+typing-extensions==3.10.0.2
+pytorch_lightning==1.5.10
+wandb==0.12.21
+DLLogger@git+https://github.com/NVIDIA/dllogger#egg=dllogger
```
we can't have this one in requirements.txt because that will break the pip build (pypi doesn't accept this kind of stuff)

tomsercu commented on requirements.txt:
```diff
@@ -0,0 +1,15 @@
+pytest
```
The whole contents of this file is actually *only* required for installing esm with esmfold; see README instructions on `add the [esmfold] option to your pip install`. This is still nice though, but let's do this:
1. rename this file to requirements_esmfold.txt
2. Move all the stuff from setup.py `extras["esmfold"]` to this file, unpin versions as much as possible
3. Make sure it's minimal, don't think we should need something that's not curently in `extras["esmfold"]` ?
4. Read this file into list for `extras["esmfold"]`

tomsercu reviewed (changes requested):
CLI scripts and naming looks good!
The added `requirements.txt` stuff needs some changes,  ideally this could be split off into a separate PR since it is a totally separate thing.
Thanks, this is starting to look nice!

tomsercu commented:
@andersoncarlosfs I'd like to merge this PR!
Can you undo the `I changed the environment.yml to use a requirements.txt` for now so we can rename and merge the CLI part?

andersoncarlosfs commented:
> @andersoncarlosfs I'd like to merge this PR!
> Can you undo the `I changed the environment.yml to use a requirements.txt` for now so we can rename and merge the CLI part?

I removed the pip options

tomsercu commented on README.md:
````diff
@@ -321,6 +334,65 @@ diverse sequences but likely with lower native sequence recovery.
+  --cpu-offload         Enable CPU offloading
+```
+
 #### Scoring sequences
````
This still belongs to the `### Inverse Folding`, should be moved up

andersoncarlosfs replied:
Done!

tomsercu commented on README.md:
```diff
@@ -321,6 +334,65 @@ diverse sequences but likely with lower native sequence recovery.
 The default sampling temperature is 1. To optimize for native sequence
 recovery, we recommend sampling with low temperature such as 1e-6.
 
+### Command-line interface <a name="cli"></a>
```
Nice, is this autogenerated from running `esm-extract -h`?

andersoncarlosfs replied:
Yep!

tomsercu commented on README.md:
````diff
@@ -226,8 +227,16 @@ Besides `esm.pretrained.esmfold_v1()` which is the best performing model we reco
-We also provide a script (`scripts/esmfold_inference.py`) that efficiently predicts structures in bulk from a FASTA file using ESMFold. This can be run with
-
+We also provide a script (`esm-fold` or `scripts/esmfold_inference.py`) that efficiently predicts structures in bulk from a FASTA file using ESMFold. This can be run with one of the following commands:
+```bash
````
We can just document the CLI version and remove the `python scripts/esmfold_inference.py` invocation.

andersoncarlosfs replied:
Done!

tomsercu reviewed (approved):
LGTM! Thanks for finishing this up.

andersoncarlosfs commented:
> After those last two small comments @andersoncarlosfs ,
> and after @nikitos9000 you take a closer look and test it with `pip install`, we can merge this.
> After merging this we can roll this into a new minor pypi release.

Okidoki!

nikitos9000 commented:
Thanks for the updates @andersoncarlosfs this looks really convenient! I had to create esm/scripts/__init__.py in order to make `import esm.scripts` work (as it's not a 'namespace' package) but otherwise I tested it and everything works good so merging it in.

andersoncarlosfs commented:
> Thanks for the updates @andersoncarlosfs this looks really convenient! I had to create `esm/scripts/__init__.py` in order to make `import esm.scripts` work (as it's not a 'namespace' package) but otherwise I tested it and everything works good so merging it in.

🙃
